app/models/piece.rb

```diff
@@ -2,4 +2,50 @@
 
 class Piece < ApplicationRecord
   belongs_to :game
+
+
+  def is_obstructed?(x, y)
+    x_sorted_array = [x, x_position].sort
+    y_sorted_array = [y, y_position].sort
```

Comment:
So if I understand this right this is just an array with two elements. The first part (x or y) is the new x and y positions?  The second part (x_position and y_position) is the old x and y positions?

Reply:
Yes, it's an array with the new position followed by the original position.